blocks/init/src/blocks/components/button/button.php

```diff
@@ -26,12 +27,14 @@
 $style_color      = isset( $button['styleColor'] ) ? "{$component_class}__color--{$button['styleColor']}" : '';
 $style_size       = isset( $button['styleSize'] ) ? "{$component_class}__size--{$button['styleSize']}" : '';
 $style_size_width = isset( $button['styleSizeWidth'] ) ? "{$component_class}__size-width--{$button['styleSizeWidth']}" : '';
+$is_anchor        = ( $is_anchor === true ) ? 'js-scroll-to-anchor' : '';
```

Comment:
If the value of `$is_anchor` is a boolean, then this will suffice:

```suggestion
$is_anchor        = $is_anchor ? 'js-scroll-to-anchor' : '';
```